src/backend/distributed/planner/relation_restriction_equivalence.c

```diff
@@ -1832,13 +1838,128 @@ FilterPlannerRestrictionForQuery(PlannerRestrictionContext *plannerRestrictionCo
 }
 
 
+/*
+ * GetRestrictInfoListForRelation gets a range table entry and planner
+ * restriction context. The function returns a list of expressions that
+ * appear in the restriction context for only the given relation. And,
+ * all the varnos are set to 1.
+ */
+List *
+GetRestrictInfoListForRelation(RangeTblEntry *rangeTblEntry,
+							   PlannerRestrictionContext *plannerRestrictionContext)
+{
+	RelationRestriction *relationRestriction =
+		RelationRestrictionForRelation(rangeTblEntry, plannerRestrictionContext);
+	if (relationRestriction == NULL)
+	{
+		return NIL;
+	}
+
+	RelOptInfo *relOptInfo = relationRestriction->relOptInfo;
+	List *joinRestrictInfo = relOptInfo->joininfo;
+	List *baseRestrictInfo = relOptInfo->baserestrictinfo;
+
+	List *joinRestrictClauseList = get_all_actual_clauses(joinRestrictInfo);
+	if (ContainsFalseClause(joinRestrictClauseList))
+	{
+		/* found WHERE false, no need  to continue */
+		return copyObject((List *) joinRestrictClauseList);
```

Comment:
What is the point of returning `joinRestrictClauseList `?

Reply:
I think the point is that if we return `NIL` then it won't contain the FALSE clause, alternatively we could just return `FALSE` but we shouldn't return `NIL` I think.